CRM/Contribute/Form/Contribution/Confirm.php

```diff
@@ -155,19 +155,19 @@ public static function getContributionParams(
     $paymentProcessorOutcome, $receiptDate, $recurringContributionID) {
     $contributionParams = [
       'financial_type_id' => $financialTypeID,
-      'receive_date' => (CRM_Utils_Array::value('receive_date', $params)) ? CRM_Utils_Date::processDate($params['receive_date']) : date('YmdHis'),
+      'receive_date' => isset($params['receive_date']) ? CRM_Utils_Date::processDate($params['receive_date']) : date('YmdHis'),
```

Comment:
Is it possible for $params['receive_date'] to be the empty string? If so then this now evaluates to null instead of today's date.

Reply:
@demeritcowboy My understanding of CRM_Utils_Array::value was that it would return the value of `receive_date` if set otherwise NULL. So I don't think this line changes anything?

Reply:
Yep that's what it does but consider when `$params['receive_date'] = "";`

It's also checkable with cv:

`cv ev "$params['receive_date'] = ''; $result = (CRM_Utils_Array::value('receive_date', $params)) ? CRM_Utils_Date::processDate($params['receive_date']) : date('YmdHis'); var_dump($result);"

vs

`cv ev "$params['receive_date'] = ''; $result = isset($params['receive_date']) ? CRM_Utils_Date::processDate($params['receive_date']) : date('YmdHis'); var_dump($result);"`

The difference would also come up if `$params['receive_date'] = 0;` or `$params['receive_date'] = FALSE;`, but that seems much less likely to happen.

Reply:
Oh, I see now. I'll update the PR :-)